File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Validators/CedulaValidator.cs
using System.Text.RegularExpressions;
using BanquitoServer_Soap_DotNet_GR01.Constants;

namespace BanquitoServer_Soap_DotNet_GR01.Validators
{
    /// <summary>
    /// Validador para cédulas ecuatorianas
    /// </summary>
    public static class CedulaValidator
    {
        /// <summary>
        /// Validar que la cédula tenga el formato correcto
        /// </summary>
        /// <param name="cedula">Cédula a validar</param>
        /// <param name="mensajeError">Mensaje de error si la validación falla</param>
        /// <returns>True si es válida, False si no</returns>
        public static bool Validar(string cedula, out string mensajeError)
        {
            mensajeError = string.Empty;

            // Validar que no sea nula o vacía
            if (string.IsNullOrWhiteSpace(cedula))
            {
                mensajeError = ErrorMessages.CedulaRequerida;
                return false;
            }

            // Validar formato (10 dígitos)
            if (!Regex.IsMatch(cedula, @"^\d{10}$"))
            {
                mensajeError = ErrorMessages.CedulaInvalida;
                return false;
            }

            return true;
        }
    }
}

File: U2/PROYECTO/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Validators/CedulaValidator.java
package ec.edu.monster.Validators;

import ec.edu.monster.Constants.ErrorMessages;
import java.util.regex.Pattern;

/**
 * Validador de cédula
 */
public final class CedulaValidator {

    private static final Pattern CEDULA_PATTERN = Pattern.compile("^\\d{10}$");

    private CedulaValidator() {
    }

    /**
     * Validar formato de cédula
     * 
     * @param cedula Cédula a validar
     * @return Resultado de validación con mensaje de error si aplica
     */
    public static ValidationResult validar(String cedula) {
        if (cedula == null || cedula.isBlank()) {
            return new ValidationResult(false, ErrorMessages.CEDULA_REQUERIDA);
        }

        if (!CEDULA_PATTERN.matcher(cedula).matches()) {
            return new ValidationResult(false, ErrorMessages.CEDULA_INVALIDA);
        }

        return new ValidationResult(true, null);
    }

    /**
     * Clase interna para resultado de validación
     */
    public static class ValidationResult {
        private final boolean valido;
        private final String mensajeError;

        public ValidationResult(boolean valido, String mensajeError) {
            this.valido = valido;
            this.mensajeError = mensajeError;
        }

        public boolean isValido() {
            return valido;
        }

        public String getMensajeError() {
            return mensajeError;
        }
    }
}
